pkg/controller/operators/catalog/operator.go

```diff
@@ -940,13 +940,26 @@ func (o *Operator) syncSubscriptions(obj interface{}) error {
 }
 
 func (o *Operator) nothingToUpdate(logger *logrus.Entry, sub *v1alpha1.Subscription) bool {
+	if sub.Status.InstallPlanRef != nil && sub.Status.State == v1alpha1.SubscriptionStateUpgradePending {
+		if o.isInstallPlanMissing(sub) {
+			fmt.Println("Testing Install Plan is missing...")
+			return false
+		}
+		logger.Debugf("skipping update: installplan already created")
+		return true
+	}
 	// Only sync if catalog has been updated since last sync time
 	if o.sourcesLastUpdate.Before(sub.Status.LastUpdated.Time) && sub.Status.State != v1alpha1.SubscriptionStateNone && sub.Status.State != v1alpha1.SubscriptionStateUpgradeAvailable {
 		logger.Debugf("skipping update: no new updates to catalog since last sync at %s", sub.Status.LastUpdated.String())
 		return true
 	}
-	if sub.Status.InstallPlanRef != nil && sub.Status.State == v1alpha1.SubscriptionStateUpgradePending {
-		logger.Debugf("skipping update: installplan already created")
+	return false
+}
+
+// isInstallPlanMissing checks if the installplan is missing or not
+func (o *Operator) isInstallPlanMissing(sub *v1alpha1.Subscription) bool {
+	_, err := o.client.OperatorsV1alpha1().InstallPlans(sub.GetNamespace()).Get(context.TODO(), sub.Status.Install.Name, metav1.GetOptions{})
+	if err != nil {
```

Comment:
An error here doesn't imply that the plan is missing (for example, the API server could be unreachable).